src/Infrastructure/Repository/TaskRepository.php

```diff
@@ -140,6 +140,30 @@ public function findById(string $taskId): ?TaskInterface
         return $task === null ? null : $this->changePhpCommand($task);
     }
 
+    /**
+     * @param array<string> $taskIds
+     *
+     * @return array<string, \SprykerSdk\SdkContracts\Entity\TaskInterface>
+     */
+    public function findByIds(array $taskIds): array
+    {
+        $criteria = [
+            'id' => $taskIds,
+        ];
+
+        /** @var array<\SprykerSdk\Sdk\Infrastructure\Entity\Task> $tasks */
+        $tasks = $this->findBy($criteria);
+        $tasksMap = array_flip($taskIds);
+
+        foreach ($tasks as $task) {
+            $tasksMap[$task->getId()] = $this->changePhpCommand($task);
+        }
+
+        return array_filter($tasksMap, function ($task) {
+            return $task instanceof TaskInterface;
+        });
```

Comment:
Doctrine can return values "indexed by field" out of the box and you don't need all the code above.

Reply:
+